canvas.html

```diff
@@ -0,0 +1,240 @@
+            const image = images[i];
+            const dx = mouseX - image.x;
+            const dy = mouseY - image.y;
+            const distance = Math.sqrt(dx * dx + dy * dy);
+            if (distance <= image.width / 2) {
+                selectedImageIndex = i;
+                offsetX = mouseX - image.x;
+                offsetY = mouseY - image.y;
+                canvas.addEventListener('mousemove', handleMouseMove);
+                canvas.addEventListener('mouseup', handleMouseUp);
+                updateSelectedImageInfo();
+                return;
+            }
+        }
+    }
+
+    function handleMouseMove(event) {
+        const mouseX = event.clientX - canvas.getBoundingClientRect().left;
+        const mouseY = event.clientY - canvas.getBoundingClientRect().top;
+        images[selectedImageIndex].x = mouseX - offsetX;
+        images[selectedImageIndex].y = mouseY - offsetY;
+        drawImages();
+    }
+
+    function handleMouseUp() {
+        canvas.removeEventListener('mousemove', handleMouseMove);
+        canvas.removeEventListener('mouseup', handleMouseUp);
+                updateSelectedImageInfo();
+    }
+
+    function handleWheel(event) {
+        event.preventDefault();
+        const delta = Math.sign(event.deltaY);
+        const rotationAmount = (delta * Math.PI) / 180 * rotationPrecision;
+        images[selectedImageIndex].rotation += rotationAmount;
+        drawImages();
+    }
+
+    function handleKeyDown(event) {
+        if (event.key === 'Backspace' && selectedImageIndex !== null && document.activeElement !== document.getElementById('searchInput')) {
+            images.splice(selectedImageIndex, 1);
+            selectedImageIndex = null;
+            drawImages();
+            updateSelectedImageInfo();
+        } else if (event.key === 'ArrowUp') {
+            moveImageUp();
+            event.preventDefault();
+        } else if (event.key === 'ArrowDown') {
+            moveImageDown();
+            event.preventDefault();
+        }
+    }
+
+    function getStickerIdsFromUrl() {
+        const params = new URLSearchParams(window.location.search)
+        const slot0 = params.get("slot0")
+        const slot1 = params.get("slot1")
+        const slot2 = params.get("slot2")
+        const slot3 = params.get("slot3")
+        const slot4 = params.get("slot4")
+
+        return [slot0, slot1, slot2, slot3, slot4]
+    }
+
+    async function findStickersById(idsToFind) {
+        
+        const response = await fetch("https://cs-sticker.com/stickers_by_id.json")
+
+        const stickersById = await response.json()
+
+        return idsToFind.map(_id=>stickersById[_id])
+
+    }
+
+    async function placeSticker() {
+        const stickerIds = getStickerIdsFromUrl();
+        const foundStickers = await findStickersById(stickerIds)
+        console.log(foundStickers)
+        const validStickers = foundStickers.filter(sticker => sticker !== null);
```

Comment:
Maybe this can be done in `findStickersById` method?

Reply:
I would prefer to keep it out of that function for now. I might have another use-case for the function where I don't need to check for validity for example.

Reply:
Sure, I know this is still WIP pull request.